logger/logger_test.go

```diff
@@ -0,0 +1,303 @@
+	assert.Len(t, mock.warnCalls, 1, "Should have one warn call")
+	assert.Len(t, mock.errorCalls, 1, "Should have one error call")
+
+	assert.Equal(t, "", mock.debugCalls[0].msg)
+	assert.Equal(t, "", mock.infoCalls[0].msg)
+	assert.Equal(t, "", mock.warnCalls[0].msg)
+	assert.Equal(t, "", mock.errorCalls[0].msg)
+
+	// Restore default logger
+	logger = NewGlogLogger()
+}
+
+func TestMultipleArguments(t *testing.T) {
+	// Initialize glog flags
+	flag.Set("logtostderr", "true")
+
+	mock := newMockLogger()
+	logger = mock
+
+	Infof("message: %s, number: %d, float: %f, bool: %v", "test", 42, 3.14, true)
+
+	assert.Len(t, mock.infoCalls, 1, "Should have one info call")
+	assert.Equal(t, "message: %s, number: %d, float: %f, bool: %v", mock.infoCalls[0].msg)
+	assert.Equal(t, []any{"test", 42, 3.14, true}, mock.infoCalls[0].args)
+
+	// Restore default logger
+	logger = NewGlogLogger()
+}
+
+func TestNoArgs(t *testing.T) {
+	// Initialize glog flags
+	flag.Set("logtostderr", "true")
+
+	mock := newMockLogger()
+	logger = mock
+
+	Infof("simple message")
+	Debugf("simple debug")
+	Warnf("simple warning")
+	Errorf("simple error")
+
+	assert.Len(t, mock.infoCalls, 1, "Should have one info call")
+	assert.Len(t, mock.debugCalls, 1, "Should have one debug call")
+	assert.Len(t, mock.warnCalls, 1, "Should have one warn call")
+	assert.Len(t, mock.errorCalls, 1, "Should have one error call")
+
+	assert.Empty(t, mock.infoCalls[0].args)
+	assert.Empty(t, mock.debugCalls[0].args)
+	assert.Empty(t, mock.warnCalls[0].args)
+	assert.Empty(t, mock.errorCalls[0].args)
+
+	// Restore default logger
+	logger = NewGlogLogger()
+}
+
+func TestWithRealGlogLogger(t *testing.T) {
+	// Initialize glog flags
+	flag.Set("logtostderr", "true")
+	flag.Set("v", "2")
+
+	// Use real GlogLogger
+	logger = NewGlogLogger()
+
+	// These should not panic
+	assert.NotPanics(t, func() {
+		Debugf("debug message")
+		Infof("info message")
+		Warnf("warning message")
+		Errorf("error message")
+	}, "Real GlogLogger should not panic")
+}
+
+func TestSpecialCharacters(t *testing.T) {
+	// Initialize glog flags
+	flag.Set("logtostderr", "true")
+
+	mock := newMockLogger()
+	logger = mock
+
+	Infof("message with special chars: \n\t\"quotes\" and 'apostrophes'")
+
+	assert.Len(t, mock.infoCalls, 1, "Should have one info call")
+	assert.Equal(t, "message with special chars: \n\t\"quotes\" and 'apostrophes'", mock.infoCalls[0].msg)
+
+	// Restore default logger
+	logger = NewGlogLogger()
+}
+
+func TestConcurrentLogging(t *testing.T) {
+	// Initialize glog flags
+	flag.Set("logtostderr", "true")
+
+	mock := newMockLogger()
+	logger = mock
+
+	done := make(chan bool)
+
+	// Test concurrent logging
+	for i := 0; i < 10; i++ {
+		go func(n int) {
+			Infof("concurrent message %d", n)
+			done <- true
+		}(i)
+	}
+
+	// Wait for all goroutines to complete
+	for i := 0; i < 10; i++ {
+		<-done
+	}
+
+	assert.Equal(t, 10, len(mock.infoCalls), "Should have 10 info calls from concurrent execution")
```

Comment:
Is this test flaky or is something wrong? The test is failing in the the validate github action workflow.